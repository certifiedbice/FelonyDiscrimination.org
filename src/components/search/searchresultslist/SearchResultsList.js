import React,{Component} from 'react'
import SearchResult from '../searchresult/SearchResult';
import './SearchResultsList.css';

export default class SearchResultsList extends Component{
	render(){
		return(
			<section id='search-results'>
				<div>{/*results: total information here*/}</div>
				{this.props.search_results.map((result,index)=><SearchResult key={index} result={result}/>)}
			</section>
		);
	}
}