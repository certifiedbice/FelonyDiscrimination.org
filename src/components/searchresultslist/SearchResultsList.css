#search-results {
    flex: 1;
    text-align: center;
}