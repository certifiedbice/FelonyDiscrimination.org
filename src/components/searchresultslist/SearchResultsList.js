import React,{Component} from 'react'
import SearchResult from '../searchresult/SearchResult';
import './SearchResultsList.css';

export default class SearchResultsList extends Component{
	render(){
		return(
			<section id="search-results">
				{this.props.search_results.map((result,index)=><SearchResult id={index} result={result}/>)}
			</section>
		);
	}
}
