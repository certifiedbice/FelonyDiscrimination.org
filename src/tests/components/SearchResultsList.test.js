import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import SearchResultsList from '../../components/searchresultslist/SearchResultsList';

const search_results=[{"id":"5","org_name":"brownstone realty ltd","org_phone":"3038328155","org_st_addr":"789 sherman st","org_city":"denver","org_state":"co","org_zipcode":"80203","org_type":"housing"}]

let x=0;

let id=()=>{x+=1;return x;}

// Need to apply a fix for this as the one present isn't working

// console.error node_modules/react/cjs/react.development.js:315       
//       Warning: Each child in a list should have a unique "key" prop.    
      
//       Check the render method of `SearchResultsList`. See https://fb.me/react-warning-keys for more information.
//           in SearchResult (at SearchResultsList.js:9)
//           in SearchResultsList (at SearchResultsList.test.js:15)        
//           in Router (created by BrowserRouter)
//           in BrowserRouter (at SearchResultsList.test.js:15)

it('renders without crashing',()=>{
    const div=document.createElement('div');
    ReactDOM.render(<BrowserRouter><SearchResultsList id={id} search_results={search_results}/></BrowserRouter>,div);
});

it('renders correctly',()=>{
    const tree=renderer
        .create(<BrowserRouter><SearchResultsList id={id} search_results={search_results}/></BrowserRouter>)
        .toJSON();
        expect(tree).toMatchSnapshot();
});