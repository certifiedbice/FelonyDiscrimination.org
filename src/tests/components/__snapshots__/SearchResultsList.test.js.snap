// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  id="search-results"
>
  <div
    className="search-result-item"
  >
    <div>
      <h2>
        <a
          href="/orgs/5"
          id="5"
          onClick={[Function]}
        >
          brownstone realty ltd
        </a>
      </h2>
    </div>
    <div>
      3038328155
    </div>
    <div>
      789 sherman st
    </div>
    <div>
      denver
      , 
      co
       
      80203
    </div>
  </div>
</section>
`;
